### PR TITLE
[CORE-652] add more test cases and comments to bridge module

### PR DESCRIPTION
**Pull request by tqin7**
### Changelist
1. add more test cases and comments to bridge module, specifically
* delay bridge completion by 0 blocks
* more failure cases of bridge tx rejection due to mismatching bridge event content
* delay msg error is propagated `AcknowledgeBridges`
* completing a bridge whose coin amount is negative
2. do not panic in `AcknowledgeBridges` if it fails to delay a complete bridge message
3. do not measure latency if there's no bridge event to acknowledge or if bridging is disabled

### Test Plan
[Describe how this PR was tested (if applicable)]

### Author/Reviewer Checklist
- [ ] If this change affects functionality (features, bug fixes, breaking changes, etc.), update `protocol/CHANGELOG.md` and/or `indexer/CHANGELOG.md` appropriately.
- [ ] If this PR has changes that result in a different app state given the same prior state and transaction list, manually add the `state-breaking` label.
- [ ] If this PR isn't state-breaking but has changes that modify behavior in `PrepareProposal` or `ProcessProposal`, manually add the label `proposal-breaking`.
- [ ] If this PR is one of many that implement a specific feature, manually label them all `feature:[feature-name]`.
- [ ] If you wish to for mergify-bot to automatically create a PR to backport your change to a release branch, manually add the label `backport/[branch-name]`.
- [ ] Manually add any of the following labels: `refactor`, `chore`, `bug`.

### CODE REVIEW

**Review by BrendanChou — APPROVED** [review]

File: protocol/x/bridge/keeper/acknowledge_bridges.go [hunk 1]
```diff
@@ -76,28 +75,33 @@ func (k Keeper) GetAcknowledgeBridges(
 	}
 }
 
-// AcknowledgeBridges acknowledges a list of bridge events.
+// AcknowledgeBridges acknowledges a list of bridge events and returns an error if any of following
+// - bridging is disabled.
+// - fails to delay a `MsgCompleteBridge` for any bridge event.
+// - fails to upate `AcknowledgedEventInfo` in state.
```
Comment (clemire) [inline protocol/x/bridge/keeper/acknowledge_bridges.go:81]:
```suggestion
// - fails to update `AcknowledgedEventInfo` in state.
```

File: protocol/x/bridge/keeper/acknowledge_bridges.go [hunk 2]
```diff
@@ -76,28 +75,33 @@ func (k Keeper) GetAcknowledgeBridges(
...
 func (k Keeper) AcknowledgeBridges(
 	ctx sdk.Context,
 	bridgeEvents []types.BridgeEvent,
 ) (err error) {
+	if len(bridgeEvents) == 0 {
```
Comment (clemire) [inline protocol/x/bridge/keeper/acknowledge_bridges.go:86]: Wasn't this part of a previous PR? Is there a test case for this below?

File: protocol/x/bridge/keeper/acknowledge_bridges.go [hunk 3]
```diff
@@ -76,28 +75,33 @@ func (k Keeper) GetAcknowledgeBridges(
...
+		return nil
+	}
+	safetyParams := k.GetSafetyParams(ctx)
```
Comment (clemire) [inline protocol/x/bridge/keeper/acknowledge_bridges.go:89]: Wasn't this disabling logic part of a previous PR? Is there a test case for this below?

File: protocol/x/bridge/keeper/acknowledge_bridges_test.go [hunk 4]
```diff
@@ -69,23 +89,25 @@ func TestAcknowledgeBridges(t *testing.T) {
...
-				require.ErrorIs(t, err, types.ErrBridgingDisabled)
-
-				// Verify that no messages were delayed.
-				mockDelayMsgKeeper.AssertNotCalled(t, "DelayMessageByBlocks")
+			if tc.expectedError != "" {
+				// Verify that error is as expected.
+				require.ErrorContains(t, err, tc.expectedError)
 
-				// Verify that AcknowledgedEventInfo was not updated.
+				// Verify that AEI was not updated.
```
Comment (clemire) [inline protocol/x/bridge/keeper/acknowledge_bridges_test.go:103]: super nit: I think the original comment was more clear.

**Review by clemire — APPROVED** [review]
Nice! Thanks for adding these!

**Pull request merged**